Formula/elftoolchain.rb

```diff
@@ -0,0 +1,149 @@
+class Elftoolchain < Formula
+  desc "Implementation of compilation tools for the ELF object format"
+  homepage "https://sourceforge.net/p/elftoolchain/wiki/Home/"
+  url "https://downloads.sourceforge.net/project/elftoolchain/Sources/elftoolchain-0.7.1/elftoolchain-0.7.1.tar.bz2"
+  sha256 "44f14591fcf21294387215dd7562f3fb4bec2f42f476cf32420a6bbabb2bd2b5"
+  license all_of: ["BSD-2-Clause", "BSD-3-Clause"]
+  head "https://svn.code.sf.net/p/elftoolchain/code/trunk"
+
+  depends_on "bmake" => :build
+  depends_on "subversion" => :build # TODO: remove in the next release
+  depends_on "libarchive" # needs headers
+
+  uses_from_macos "bison" => :build
+  uses_from_macos "flex" => :build
+  uses_from_macos "m4" => :build
+
+  # TODO: add conflicts after merge
+  # on_macos do
+  #   conflicts_with "libelf", because: "both install `libelf`"
+  #   conflicts_with "dwarfutils", because: "both install `libdwarf`"
+  # end
```

Comment:
Currently chose to link on macOS and keep keg-only on Linux. All packages that may conflict:
- `binutils` - keg-only on macOS
- `elfutils` - Linux-only
- `libelf` - deprecated and planning to disable/remove soon
- `mono` - due to `msc` binary, but I added workaround to only install into `libexec` and not link to `bin`. Alternatively, could just mark conflict.
- `dwarfutils` - only maintained formula that actually conflicts in current state. 

---

Another option is to install `libelf*`/`libdwarf*` into a subdirectory like `LIBDIR=#{lib}/#{name}` which may remove conflict with `libelf` & `dwarfutils`, but this may make it trickier to use by dependents/users.